pkg/agent/testdata/AKSUbuntu1604+Containerd/line70.sh

```diff
@@ -389,6 +389,23 @@ ensureDHCPv6() {
     retrycmd_if_failure 120 5 25 modprobe ip6_tables || exit $ERR_MODPROBE_FAIL
 }
 
+getPrimaryNicIP() {
+    local sleepTime=1
+    local maxRetries=10
+    local i=0
+    local ip=""
+
+    while [[ $i -lt $maxRetries ]]; do
```

Comment:
you can probably just reuse this instead: https://github.com/Azure/AgentBaker/blob/0515c7aee97fc172541691a715ce779ca87cb8e5/parts/linux/cloud-init/artifacts/cse_helpers.sh#L160

Reply:
that function does not support return value (`echo $ip` in our case), so I implemented similar retry logic.

Reply:
I think you can just have your `getPrimaryNicIP` curl IMDS, check for errors, and echo the result without the retry logic

then call `getPrimaryNicIP` using `retrycmd_if_failure`

Reply:
looks like it will print some help messages, in this case, we'll need to parse the result, it will make the code more complex, we may not want this?

```
$ result=$(retrycmd_if_failure 20 5 25 echo "aaa")
$ echo $result
aaa Executed "echo aaa" 1 times
```

Reply:
yeah... I really dislike the pattern of "returning" values via echoing to stdout

regardless, I don't want to endlessly block you on this - we can keep what you had originally

Reply:
you could also theoretically define a new helper called something like `retrycmd_if_failure_no_output` or something, which does the same thing but just drops the extra echo statements

will leave up to you